src/main/java/spoon/MavenLauncher.java

```diff
@@ -108,656 +125,144 @@ public MavenLauncher(String mavenProject, String m2RepositoryPath, SOURCE_TYPE s
-			for (TreeDependency dep : dependencyList) {
-				File file = dep.getTopLevelJar();
-				if (null != file && !addedDep.contains(dep)) {
-					addedDep.add(dep);
-					output.add(file);
-				}
-			}
-			return output;
-		}
-
-		private File getTopLevelJar() {
-			if ("pom".equals(type)) {
-				return null;
-			}
-			if (groupId != null && version != null) {
-				String fileName = artifactId + "-" + version;
-				Path depPath = Paths.get(m2RepositoryPath, groupId.replaceAll("\\.", "/"), artifactId, version);
-				File depFile = depPath.toFile();
-				if (depFile.exists()) {
-					File jarFile = Paths.get(depPath.toString(), fileName + ".jar").toFile();
-					if (jarFile.exists()) {
-						return jarFile;
-					} else {
-						LOGGER.log(Level.ERROR, "Jar not found at " + jarFile);
-					}
-				} else {
-					LOGGER.log(Level.ERROR, "Dependency not found at " + depPath);
-				}
-			}
-			return null;
-		}
-
-		void removeDependency(String groupId, String artifactId) {
-			for (TreeDependency dep : new ArrayList<>(dependencies)) {
-				if (dep.groupId != null && dep.groupId.equals(groupId) && dep.artifactId != null && dep.artifactId.equals(artifactId)) {
-					this.dependencies.remove(dep);
-				} else {
-					dep.removeDependency(groupId, artifactId);
-				}
-			}
-		}
-
-		@Override
-		public boolean equals(Object o) {
-			if (this == o) {
-				return true;
-			}
-			if (o == null || getClass() != o.getClass()) {
-				return false;
-			}
-			TreeDependency that = (TreeDependency) o;
-			return Objects.equals(groupId, that.groupId)
-					&& Objects.equals(artifactId, that.artifactId);
-		}
-
-		@Override
-		public int hashCode() {
-			return Objects.hash(groupId, artifactId);
-		}
-
-		@Override
-		public String toString() {
-			StringBuilder sb = new StringBuilder();
-			sb.append(groupId);
-			sb.append(":");
-			sb.append(artifactId);
-			sb.append(":");
-			sb.append(version);
-			if (!dependencies.isEmpty()) {
-				sb.append(" {\n");
-				for (TreeDependency dep : dependencies) {
-					String child = dep.toString();
-					for (String s : child.split("\n")) {
-						sb.append("\t");
-						sb.append(s);
-						sb.append("\n");
-					}
-				}
-				sb.append("}");
-			}
-			return sb.toString();
+	private static void generateClassPathFile(File pom, File mvnHome, SOURCE_TYPE sourceType) {
+		//Run mvn dependency:build-classpath -Dmdep.outputFile="spoon.classpath.tmp"
+		//This should write the classpath used by maven in spoon.classpath.tmp
+		InvocationRequest request = new DefaultInvocationRequest();
+		request.setPomFile(pom);
+		request.setGoals(Arrays.asList("dependency:build-classpath"));
+		Properties properties = new Properties();
+		if (sourceType == SOURCE_TYPE.APP_SOURCE) {
+			properties.setProperty("includeScope", "runtime");
+		}
+		properties.setProperty("mdep.outputFile", "spoon.classpath.tmp");
+		request.setProperties(properties);
+
+		//FIXME Should the standard output made silent and error verbose?
```

Comment:
It might be interesting to use a logger debug for this one? WDYT?

Reply:
Ok. I'll add it ASAP.